src/Lightweight/SqlQuery/MigrationPlan.hpp

```diff
@@ -52,6 +53,134 @@ using SqlColumnTypeDefinition = std::variant<
 >;
 // clang-format on
 
+namespace detail
+{
+
+template <typename T>
+constexpr bool AlwaysFalse = false;
+
+template <typename>
+struct SqlColumnTypeDefinitionOf;
+
+template <>
+struct SqlColumnTypeDefinitionOf<bool>
+{
+    static constexpr auto value = SqlColumnTypeDefinitions::Bool {};
+};
+
+template <>
+struct SqlColumnTypeDefinitionOf<char>
+{
+    static constexpr auto value = SqlColumnTypeDefinitions::Char { 1 };
+};
+
+template <>
+struct SqlColumnTypeDefinitionOf<SqlDate>
+{
+    static constexpr auto value = SqlColumnTypeDefinitions::Date {};
+};
+
+template <>
+struct SqlColumnTypeDefinitionOf<SqlDateTime>
+{
+    static constexpr auto value = SqlColumnTypeDefinitions::DateTime {};
+};
+
+template <>
+struct SqlColumnTypeDefinitionOf<SqlTime>
+{
+    static constexpr auto value = SqlColumnTypeDefinitions::Time {};
+};
+
+template <size_t Precision, size_t Scale>
+struct SqlColumnTypeDefinitionOf<SqlNumeric<Precision, Scale>>
+{
+    static constexpr auto value = SqlColumnTypeDefinitions::Decimal { .precision = Precision, .scale = Scale };
+};
+
+template <>
+struct SqlColumnTypeDefinitionOf<SqlGuid>
+{
+    static constexpr auto value = SqlColumnTypeDefinitions::Guid {};
+};
+
+template <>
+struct SqlColumnTypeDefinitionOf<int16_t>
+{
+    static constexpr auto value = SqlColumnTypeDefinitions::Smallint {};
+};
+
+template <>
+struct SqlColumnTypeDefinitionOf<uint16_t>
+{
+    static constexpr auto value = SqlColumnTypeDefinitions::Smallint {};
+};
```

Comment:
Instead of two separate entries better to create specialization like:
```suggestion

template <typename T>
requires one_of<T, uint16_t, int16_t>
struct SqlColumnTypeDefinitionOf<T>
{
    static constexpr auto value = SqlColumnTypeDefinitions::Smallint {};
};
```

Reply:
using `same_as... || ...` for now, because OneOf<> needs to be moved first to a more central helper file (tbd later).

Reply:
Done. Moved `OneOf<T, U...>` to Utils.hpp and rewrote the bits here.